### PR TITLE
Tend Wounds and Debride Infection surgeries now estimate how long you have left to finish

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
Treating an injured patient with Tend Wounds can take anywhere from a few seconds to upwards of a few minutes depending on how badly they're hurt, if you're treating them through their clothes, if they're dead, what level surgery you have, etc etc. This PR makes it so the message from succeeding the looping step gives you a little blurb about how many times you'll have to repeat it to finish up, so you can get a sense for if it's worth continuing the surgery, or if you should pursue another, quicker course of treatment.

[](https://i.imgur.com/us1CbL5.png)

Having a health analyzer in your offhand will give you updates on the precise numbers of damage left to treat, in case you want to know when you're approaching defibbable levels or you just want more precise numbers.

[](https://i.imgur.com/rH3qmOJ.png)

The specific numbers and thresholds might need some tweaking, I figured that having to repeat the same step 15+ times is a good cap for saying "listen, maybe you should just use some sutures or something", but let me know if different numbers would work better. 

I also tweaked some of the wording related to burn wounds to be a bit less unnecessarily gross.

## Why It's Good For The Game
Is your Tend Wounds surgery dragging on forever? Did you forget to open up the operating computer before you started, or do you not have one available, so you can't see how fast you're progressing? Afraid that you damned yourself to spend the next five minutes standing still, but you don't want to interrupt the treatment by using a health analyzer to check your progress? This is the PR for you!

## Changelog
:cl: Ryll/Shaps
qol: Tend Wounds and Debride Infected Flesh surgeries now tell the surgeon roughly how many more repetitions are left before the patient is fully treated.
qol: Having a health analyzer in your offhand during Tend Wounds will give you precise updates on how much damage is left after each step.
/:cl:

### CODE REVIEW

**Comment by D4C-420** [comment]
add the same thing to blood filter/ slash stomach pump. Its not fun having to scan someone who had 0.2 u of a poison left in them because the surgery takes ages .

File: code/modules/surgery/healing.dm [hunk 1]
```diff
@@ -124,6 +130,31 @@
 	desc = "A surgical procedure that provides experimental treatment for a patient's brute traumas. Heals considerably more when the patient is severely injured."
 
 /********************BRUTE STEPS********************/
+/datum/surgery_step/heal/brute/get_progress(mob/user, mob/living/carbon/target, brute_healed, burn_healed)
+	if(!brute_healed)
+		return
+
+	var/estimated_remaining_steps = target.getBruteLoss() / brute_healed
```
Comment (ATH1909) [inline code/modules/surgery/healing.dm:137]: this really should be telling them how close you're getting to bringing the patient into defib range, imo

**Comment by cacogen** [comment]
weird to think this game could actually be considerate of players' time

**Comment by Ryll-Ryll** [comment]
Hm, I want to go with ath's idea for telling you when a body can be defibbed, just need to figure out the best way to do so if possible

**Comment by Ryll-Ryll** [comment]
I didn't want to make the normal messages any more cluttered to fit in Ath's suggestion, so holding a health analyzer in your offhand will now provide numbers for how much damage is left after each step. Added a photo to the OP, and here's another of the mixed tend wounds

[](https://i.imgur.com/zttVKUf.png)

**Comment by Mothblocks** [comment]
his idea huh

File: code/modules/surgery/healing.dm [hunk 2]
```diff
@@ -124,6 +130,34 @@
...
+	var/progress_text
+
+	if(locate(/obj/item/healthanalyzer) in user.held_items)
```
Comment (Mothblocks) [inline code/modules/surgery/healing.dm:140]: this is a clever solution

**Pull request merged**